### PR TITLE
add support for apache airflow and add kms-key filter

### PR DESCRIPTION
**Pull request by vkubyshko**
This PR adds support for AWS Apache Airflow (MWAA) resources. This also provides an action to tag mwaa resources and a kms-key filter.

### CODE REVIEW

**Comment by JohnHillegass** [comment]
Thanks for the PR @vkubyshko. Could you accept the CLA? Once accepted, CI checks will begin to run and then we can begin to work on reviewing your contribution to the project.

https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/14202841/52837350/6823/#/?version=2

File: c7n/resources/mwaa.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+from c7n.manager import resources
+from c7n.query import ConfigSource, DescribeSource, QueryResourceManager, TypeInfo
+from c7n.filters.kms import KmsRelatedFilter
+from c7n.tags import RemoveTag, Tag, TagDelayedAction, TagActionFilter
+
+@resources.register('mwaa')
```
Comment (kapilt) [inline c7n/resources/mwaa.py:8]: probably want to rename to airflow

File: c7n/resources/airflow.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    class resource_type(TypeInfo):
+        service = 'mwaa'
+        id = name = 'Name'
+        enum_spec = ('list_environments', 'Environments', None)
+        detail_spec = ('get_environment', 'Name', None, 'Environment')
+        arn = 'Arn'
+        arn_type = 'environment'
+        cfn_type = 'AWS::MWAA::Environment'
+
+    permissions = (
```
Comment (PratMis) [inline c7n/resources/airflow.py:20]: Looks like the permissions should be as per the docs: https://docs.aws.amazon.com/mwaa/latest/userguide/access-policies.html
```
airflow:ListEnvironments,
airflow:GetEnvironment
```

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Pull request merged**